src/Kafka/CreateKafkaTopic/components/CreateTopicWizard.tsx

```diff
@@ -0,0 +1,183 @@
+  const { t } = useTranslation(["create-topic", "common"]);
+
+  const [topicNameValidated, setTopicNameValidated] = useState<
+    "error" | "default"
+  >("default");
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [invalidText, setInvalidText] = useState<string>("");
+  const [topicData, setTopicData] = useState<NewTopic>(initialFieldsValue);
+  const [warningModalOpen, setWarningModalOpen] = useState<boolean>(false);
+
+  const closeWizard = () => {
+    onCloseCreateTopic && onCloseCreateTopic();
+  };
+
+  const steps: WizardStep[] = [
+    {
+      name: t("topic_name"),
+      enableNext:
+        topicData?.name.trim() !== "" && topicNameValidated === "default",
+      component: (
+        <StepTopicName
+          newTopicData={topicData}
+          onTopicNameChange={setTopicData}
+          topicNameValidated={topicNameValidated}
+          onValidationCheck={setTopicNameValidated}
+          invalidText={invalidText}
+          setInvalidText={setInvalidText}
+        />
+      ),
+    },
+    {
+      name: t("partitions"),
+      canJumpTo: topicData?.name.trim() !== "",
+      component: (
+        <StepPartitions
+          newTopicData={topicData}
+          onPartitionsChange={setTopicData}
+          availablePartitionLimit={availablePartitionLimit}
+        />
+      ),
+    },
+    {
+      name: t("message_retention"),
+      canJumpTo: topicData?.name.trim() !== "",
+      component: (
+        <StepMessageRetention
+          newTopicData={topicData}
+          onChangeMessageRetention={setTopicData}
+        />
+      ),
+    },
+    {
+      name: t("replicas"),
+      canJumpTo: topicData?.name.trim() !== "",
+      component: (
+        <StepReplicas
+          replicationFactor={topicData.replicationFactor}
+          minInSyncReplica={topicData.minInSyncReplica}
+          isMultiAZ={topicData.isMultiAZ}
+        />
+      ),
+      nextButtonText: t("finish"),
+    },
+  ];
+
+  const title = t("wizard_title");
+
+  const onValidate: IWizardFooter["onValidate"] = (onNext) => {
+    if (topicData?.name.length < 1) {
+      setInvalidText(t("common:required"));
+      setTopicNameValidated("error");
+    } else {
+      setIsLoading(true);
+
+      checkTopicName(topicData?.name)
+        .then((value) =>
+          value == false
+            ? (setInvalidText(t("already_exists", { name: topicData?.name })),
+              setTopicNameValidated("error"))
+            : onNext()
+        )
+        .finally(() => setIsLoading(false));
+    }
+  };
+  const onSaveTopic = () => {
+    if (topicData.numPartitions >= availablePartitionLimit)
+      setWarningModalOpen(true);
+    else onSave(topicData);
+  };
+  return (
+    <>
+      {isSwitchChecked ? (
+        <>
+          <Divider className="kafka-ui--divider--FlexShrink" />
+          <PageSection
+            variant={PageSectionVariants.light}
+            hasOverflowScroll={true}
+          >
+            {/*  <TopicAdvanceConfig
```

Comment:
why do we have this commented code ? remove if not needed

Reply:
I'll be pushing this component in the next commit